### DISCUSSION TITLE
Experimental Offline Standalone Mac Installer for Stable Diffusion Web UI (unofficial) 20230913

### DISCUSSION THREAD

**Discussion by brkirch** [brkirch/stable-diffusion-webui#40] [Announcements]
This is an offline standalone installer/application for a modified version of AUTOMATIC1111's Stable Diffusion web UI that includes experimental changes and uses a release candidate PyTorch build for 2.1.0. Note that this is an test build so crashes and bugs should be expected. All dependencies needed to start using Stable Diffusion (other than the model) are included and this does not require Xcode command line tools to be installed.

This release currently only has the installer and application for Apple silicon Macs, an installer and application for Intel Macs will be included on a later release. macOS 13.5 or later is highly recommended, older versions of macOS are untested.

Now it is possible to run web UI as an application (which utilizes [Electron](https://www.electronjs.org)), with [an icon courtesy of @ParityError](https://github.com/AUTOMATIC1111/stable-diffusion-webui/discussions/2901?sort=new#discussioncomment-5166168). Note that the installer can create the application, so if the installer is downloaded then the app download is not needed.

When run as an application, a log file is created in `/tmp/Stable Diffusion Web UI Logs` which stores the output from the Terminal - and it is also possible to close the Terminal window or quit Terminal entirely without closing out the web UI. If you want to return the output to a Terminal window, just enter `screen -r`. When the application with web UI is quit, the process for web UI will exit as well. Models and configuration files are stored in the `Stable Diffusion Web UI` folder in the `Documents` folder in the current user’s home folder

To use Electron from a folder install, it is now necessary to use `--electron --electron-path bin-deps/Electron.app/Contents/MacOS/Electron` because Node.js is not included (since it wasn't needed other than to get Electron's path).

<hr /><em>This discussion was created from the release <a href='https://github.com/brkirch/stable-diffusion-webui/releases/tag/20230913_experimental'>Experimental Offline Standalone Mac Installer for Stable Diffusion Web UI (unofficial) 20230913</a>.</em>

**Comment by birdwatcher1** [comment]
Lora, Hypernetworks, etc, data directories not being read on UI due to folder structure (spaces in folders name) in App mode. Does work with alternate data dir when using flags, --cktp-dir , --lora-dir, etc, 

Also, getting some errors when using Interrogate CLIP about folders being read-only:

OSError: [Errno 30] Read-only file system: '/private/var/folders/7k/19mph2gj0019gl6848djvpsm0000gn/T/AppTranslocation/536627FD-A88F-40B4-ABD1-C4C17AA75DE2/d/Stable Diffusion Web UI.app/Contents/Resources/stable-diffusion-webui/models/transformers/models--bert-base-uncased'

**Reply by brkirch** [reply]
Thanks for reporting. I've fixed the issue with interrogate CLIP in [20230915_experimental](https://github.com/brkirch/stable-diffusion-webui/releases/tag/20230915_experimental).

The issue with Loras, hypernetworks, etc. not working I however am not seeing. Please post any further details you can provide (tracebacks or any related output, or an example of a folder path that isn't working). The spaces shouldn't affect it working because the data directory is a single argument (internally it is quoted, but the output doesn't show that). You may need to refresh the checkpoint, lora, etc. lists manually in the UI, and if the model is not showing as loaded despite the output saying in Terminal that it is then you may need to use command-R to refresh the application window.

**Reply by alexaaaaaander** [reply]
Were you able to work through the directory hangup?? I've been trying to set a custom path to my models, hypernetworks, embeddings, loras, etc on my local google drive folders and am having zero luck.

**Reply by jrittvo** [reply]
Just a long shot, but do the paths contain “spaces”?  Like `/Users/My Name/Folder`?  Spaces can be read as the end of the path in some instances, when they aren’t.

**Reply by alexaaaaaander** [reply]
The do have spaces, but not by choice.. **/Users/name/Library/CloudStorage/GoogleDrive-xxyyzz@gmail.com/My Drive/SD_Folders/Models**. Google forces **My Drive** in the file structure, despite all attempts to change it. I tried using single-quotes for the path to make it work, but still no luck: '/Users/My Name/Folder'

**Reply by jrittvo** [reply]
I wonder if it would work if you made a symlink to that folder on a path that had no spaces?  If it would follow the simlink, which would essentially be a placeholder for for real path?  Like make a symlink `/Users/Documents/models.lnk` pointing to the GoogleDrive folder location?  Pretty bass ackwards but you never know . .

**Comment by byewind1** [comment]
I don‘t konw how to reset web UI, who can tell me?

**Reply by DetritusMidas** [reply]
You can either go to the "Settings" tab and click the "Reload UI" button. Or you can click "View" in the macOS menu bar and click "Reload" or "Force Reload". Or you can close out of the app, then reopen it. Hope this helps.
